### DISCUSSION TITLE
Some questions about UnPackaged apps

### DISCUSSION THREAD

**Discussion by ghost1372** [microsoft/WindowsAppSDK#1654] [Q&A] [answered]
Hi, I have some questions about unPackaged apps

~~1.Can we Run apps with double click on exe file?~~ Yes
2.Currently using the registry is Virtualized, Is it also virtual in the unpackaged app?
3.The [doc](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=csharp%2Ccsharp-unpackaged#instructions-for-winui-3-unpackaged-desktop-apps) says that we must delete the **package.appxmanifest**

> 4.Delete package.appxmanifest from project.

If so, how do we perform various operations such as registering the context menu?

**Comment by andrewleader** [comment, marked as answer]
Hey @ghost1372!

1. Yes (as you answered)
2. No, nothing is virtualized in unpackaged apps
3. To register context menu, you'll need to use classic desktop APIs to do so (similar to how you would in a classic WPF app that doesn't use MSIX). Let us know if you need some help with that!

**Reply by ghost1372** [reply]
Thank you. As far as I know, we need to use the package.appxmanifest to register the context menu in Windows 11. (For adding new item in windows 11 new context menu)

**Reply by andrewleader** [reply]
Good question, I'm reaching out internally to find out about Win11 and context menus.

**Reply by andrewleader** [reply]
@ghost1372, one follow-up question: **What exactly are you trying to accomplish with the context menu extension?**

Are you just trying to have your app appear in the "Open with" menu? If you register as a handler for the given file type, you'll appear there.

Or are you trying to add custom actions like 7Zip "Extract files"/etc?

**Reply by ghost1372** [reply]
Both 😅

**Reply by andrewleader** [reply]
Alright I got the official answer:

* For "Open with", you can just register as a handler for the desired file types
* For custom context menu actions, you'll need to [use sparse packages](https://docs.microsoft.com/windows/apps/desktop/modernize/grant-identity-to-nonpackaged-apps) to be able to declare those context menu actions in your manifest. Unfortunately, it's pretty painful to use sparse packages today, [here's a blog post that should help you some](https://andrewleader.medium.com/creating-a-sparse-signed-win32-app-package-9809320cfaab). I'm not sure whether we've tested sparse + WinUI 3 unpackaged together either (something on my to-do list next week).

**Reply by ghost1372** [reply]
Thank you ❤️, I will try
